### DISCUSSION TITLE
Reusable workfow : use current branch

### DISCUSSION THREAD

**Discussion by gbetous-swir** [community/community#25714] [Actions]
<p>Hi,</p>
<p>I’m working with reusable workflows, and basically I’ve got a <code>main.yml</code> file which calls different others YML like <code>test.ym</code>l or <code>build.yml</code></p>
<pre><code class="lang-auto">jobs:
  apps:
    name : Build apps
    uses: xxx/yyy/.github/workflows/apps.yml@master

  ut:
    name : Unit tests
    uses: xxx/yyy/.github/workflows/ut.yml@master

  build:
    name: Build
    uses: xxx/yyy/.github/workflows/build.yml@master

  test:
    needs: build
    name: Integration Tests
    uses: flowthings/brkedgepkg/.github/workflows/test.yml@master
</code></pre>
<p>My issue is when I’m working on the workflow itself, I create a branch and then I have to :</p>
<ul>
<li>modify the <code>main.ym</code>l to use <code>@my-branch</code> to include my modifications</li>
<li>test my modifications</li>
<li>modify it to put back <code>uses: @master</code> before merging into master.</li>
</ul>
<p>Is there a way to be generic and say in <code>main.yml</code> “use the test.yml file of the current branch” ?</p>
<p>Thanks !</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>There is, since Tuesday:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/changelog/2022-01-25-github-actions-reusable-workflows-can-be-referenced-locally/">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/5814919/181092785-0b8231f8-2877-489b-a484-a0aea30888ec.png" class="site-icon" width="32" height="32">

      <a href="https://github.blog/changelog/2022-01-25-github-actions-reusable-workflows-can-be-referenced-locally/" target="_blank" rel="noopener">The GitHub Blog</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://github.blog/wp-content/uploads/2022/01/Product-Cloud-Enterprise@2x.png?fit=2400%2C1260" class="thumbnail" width="690" height="362"></div>

<h3><a href="https://github.blog/changelog/2022-01-25-github-actions-reusable-workflows-can-be-referenced-locally/" target="_blank" rel="noopener">GitHub Actions: Reusable workflows can be referenced locally |</a></h3>

  <p>GitHub Actions: Reusable workflows can be referenced locally</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Unfortunately there seems to be a bug when using that new syntax with the <code>pull_request</code> trigger, but I’ve successfully used it with <code>push</code>.</p><aside class="quote quote-modified" data-post="60" data-topic="203690">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/ref-head-in-reusable-workflows/203690/60">Ref Head in Reusable Workflows</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    Can confirm your issues from our end <a class="mention" href="/bcowdery">@bcowdery</a>, but without GitHub Enterprise so that’s probably irrelevant for this error. 
This is the workflow call: 
call-setup:
    uses: ./.github/workflows/&lt;workflow&gt;.yml

…which yields this error: 
The workflow is not valid. &lt;org&gt;/&lt;repo&gt;/.github/workflows/&lt;workflow&gt;.yml@e302&lt;rest of sha&gt;: Expected document start parse event Unexpected type '' encountered while reading 'root'. The type 'MappingToken' was expected.

Potentially interesting: the SHA that’s be…
  </blockquote>
</aside>

**Comment by gbetous-swir** [comment]
<p>Thank you for your fast answer !</p>
<p>Will it be an issue when merging into master as I use a Pull Request mechanism from my development branch ?</p>

**Comment by airtower-luna** [comment]
<p>Take a look at the discussion in that other thread, from what it says there calling workflows with <code>on: pull_request</code> break. I don’t see why using PRs otherwise should be affected, unless of course the bug prevents required checks from running.</p>
